[18.0][ADD] account_payment_sepa_base (ex account_banking_pain_base)

alexis-via opened the pull request.

Commit:
[ADD] account_banking_pain_base: Akretion-Noviat code sprint dated 2013-11-21:

- add module account_banking_pain_base
- add support for initiating party identification, priority and structured remittance info in XML file
- the requested execution date now uses the fields date_prefered and date_scheduled (the field of the wizard has been removed)
- the 'convert to ascii' feature is now an option of the payment mode (enabled by default)
- set 'communication' field of payment.line to 140 chars and hide field 'communication2'
- Code factoring for SCT and SDD
- Add support for direct debit migration from national format to SEPA
- Source : Standard-XML-SDD-Initiation-v3-EN by Febelfin
- Add group in order to mask the fields for "Original Mandate Indentification" for users in countries that don't required it.
- Add tracking on important fields of the mandate.
- Better form view on company.
- Update constraint on mandate following my devs on "Original Mandate Identification".
- Add support for Party Identifier for Belgium. Can be easily extended for other countries.
- Mutualize more code between SCT and SDD.

Commit:
Launchpad automatic translations update.

acysos committed:
[FIX] account_banking_pain_base: support party identifier for Spain

pedrobaeza committed:
[MIG] account_banking_pain_base: Migration to 8.0

sbidoul committed:
[IMP] account_banking_pain_base: Several things:

* Make it easier to extend structured communication types.
* Use new style to make selection extendable.
* Help text for bank and BIC fields. Courtesy of Alexis de Lattre.
* Do not use InstrPrty for Direct Debit.

Commit:
Use attachments instead of creating a specific object for each kind of payment order

phschmidt committed:
Add requested  descriptions in an OCA README.rst file

Commit:
Fix bug #111 New field initiating_party_identifier on res.company (migration script included)

alexis-via committed:
[FIX+IMP] account_banking_pain_base: Port to new API fix + Add bank.payment.lines object to allow grouping in the payments

pedrobaeza committed:
[IMP] *: Short headers + Bump version numbers + es translations

antespi committed:
[IMP] Define SEPA identifiers per payment mode

alexis-via committed:
[MIG] account_banking_pain_base: Migration to 9.0

* Add support for non-SEPA DD and CT without without IBANs
* Add hooks for l10n_ch_sepa (pain.001.001.03.ch.02)
* Move code of account_banking_pain_base/models/banking_export_pain.py to account_banking_pain_base/models/account_payment_order.py
* Add postal address in party block
* Adds flexibility to support direct debit direct debit for switzerland (pain.008.001.02.ch.01)
* Implement SEPA Core recommandations for postal address
* Add field local_instrument in payment lines and bank payment lines
* Add the bank_line as argument to some methods that generate SEPA XML

Commit:
Port almost all modules to v10 (#305)

Port almost all modules to v10

* Update to EPC Rulebook v9.2 that start to apply on 2016-11-20 (bug #300)

alexis-via committed:
Add support for ISO20022 "Category Purpose"

Add local instrument 'INST' for SEPA Instant Credit Transfer

pedrobaeza committed:
[IMP] account_banking*: Show identifiers at payment mode level

Now the visibility is controlled through a security group.

etobella committed:
[MIG] account_banking_pain_base

andreparames committed:
[ADD] sepa: support Initiating Party Scheme

sbidoul committed:
[ADD] sepa: support for Purpose Code

StefanRijnhart committed:
[ADD] Allow for multiple SEPA payment methods with different versions (#493) (#497)

sbidoul committed:
[IMP] pain: add hook for party identifiers + improve Postal Address block

Use TwnNm for city and PstlCd for zip.

alessandrocamilli committed:
Added translation using Weblate (Italian)

Currently translated at 9.6% (8 of 83 strings)

Translation: bank-payment-11.0/bank-payment-11.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-11-0/bank-payment-11-0-account_banking_pain_base/it/

Commit:
PstCd and TwnNm not present in pain files 001.003.xx and 008.003.xx. Added check of pain_flavor.

sergio-teruel committed:
[12.0][MIG] account_banking_pain_base: Migration to v12.0

pedrobaeza committed:
[MIG] account_banking_pain_base: Migration to 12.0

* Standard migration procedure
* README by fragments
* Fixed tests
* Change
* Fix related no readonly in account_banking_pain_base
* Override default partner bank account check on account_payment_partner
* Convert Python date to string

ThomasBinsfeld committed:
[REF] Account Banking Pain Base: constraint on BIC
Prevent inserting BIC not respecting the SEPA pain schemas.

Commit:
[11.0] account_banking_pain_base: Allow override of address block

TDu committed:
[12.0][FIX] account_banking_pain_base bic constrains

The bic fields is not required, so the constraint must take that into
consideration and accept empty bic.

pedrobaeza committed:
[FIX] account_banking_pain_base: flake8

Commit:
Translated using Weblate (Italian)

Currently translated at 32.2% (57 of 177 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_banking_pain_base/it/

SimoRubi committed:
[IMP] account_banking_pain_base: Provide a more detailed message when _prepare_field fails

CarlesAntoli committed:
Added translation using Weblate (Catalan)

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 11.5% (21 of 182 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_banking_pain_base/sl/

ValentinVinagre committed:
[IMP] account_banking_pain_base: black, isort, prettier

ValentinVinagre committed:
[MIG] account_banking_pain_base: Migration to 13.0

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (171 of 171 strings)

Translation: bank-payment-13.0/bank-payment-13.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-13-0/bank-payment-13-0-account_banking_pain_base/es_AR/

robinkeunen committed:
[IMP] ab_pain_base: use account holder name in party block

CarlosRoca13 committed:
[MIG] account_banking_pain_base: Migration to v14

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 45.6% (78 of 171 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_banking_pain_base/nl/

alexis-via committed:
Improve computation of sepa on account.payment.order: check IBAN is in SEPA zone

Update move line generation to get transfer account from bank journal
Update payment mode configuration accordingly (3 fields removed)
Several improvements in payment order tree and form view

alexis-via committed:
account_payment_partner: add optional="hide" on payment mode in invoice tree view

alexis-via committed:
Update test suite

Fix bug in account_banking_mandate when a res.partner.bank is not attached to a particular company (company_id is NOT required on res.partner.bank)

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 70.7% (121 of 171 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_banking_pain_base/fr/

simonstromb committed:
Added translation using Weblate (Swedish)

Currently translated at 2.8% (5 of 174 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_banking_pain_base/sv/

manuelregidor committed:
[MIG]account_banking_pain_base: Migration to 15.0

jabelchi committed:
Added translation using Weblate (Catalan)

Currently translated at 2.9% (5 of 171 strings)

Translation: bank-payment-15.0/bank-payment-15.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-15-0/bank-payment-15-0-account_banking_pain_base/ca/

ramiadavid committed:
[FIX] account_banking_pain_base, account_banking_sepa_credit_transfer, account_banking_direct_debit: Remove Environment.manage()

Environment.manage() is deprecated in v 15.0

pedrobaeza committed:
[REF] account_banking_pain_base: Adapt module to native payment refactoring

luc-demeyer committed:
[FIX]generate pain address block that is supported by german banks

pedrobaeza committed:
[FIX] account_banking_pain_base: Proper condition for AdrLine

Steps to reproduce the problem:

- Have a partner with no zip code and no city.
- Include it in a payment order.
- Generate the SEPA XML file.

Current behavior: The file couldn't be generated because there's an
empty <AdrLine> element.

We put properly parenthesis in the conditions for this to no happen.

TT43260

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 31.5% (54 of 171 strings)

Translation: bank-payment-15.0/bank-payment-15.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-15-0/bank-payment-15-0-account_banking_pain_base/it/

BT-anieto committed:
[15.0] [FIX] account_payment_order: Update list of IBAN codes.

Update list of IBAN codes. Added Austria (AT) and Andorra (AD)

pedrobaeza committed:
[MIG] account_banking_pain_base: Migration to 16.0

Commit:
[UPD] Update account_banking_pain_base.pot

OCA-git-bot committed:
[UPD] README.rst

Gelo-fl committed:
Translated using Weblate (Spanish)

Currently translated at 64.7% (110 of 170 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_pain_base/es/

Gelo-fl committed:
Translated using Weblate (Spanish)

Currently translated at 98.8% (168 of 170 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_pain_base/es/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 60.0% (102 of 170 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_pain_base/hr/

remi-filament committed:
Translated using Weblate (French)

Currently translated at 100.0% (170 of 170 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_pain_base/fr/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 31.1% (53 of 170 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_pain_base/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (170 of 170 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_pain_base/es/

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Translated using Weblate (Italian)

Currently translated at 31.1% (53 of 170 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_pain_base/it/

alexis-via committed:
Add warning when payment order is not SEPA

The field 'sepa' on account.payment.order is only display for SEPA
payment methods.
If the option "show warning if not SEPA" is enabled on the payment
method, a warning banner is now displayed on payment orders with a SEPA
payment method which are not SEPA.

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_pain_base/

Commit:
[UPD] Update account_banking_pain_base.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_pain_base/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (175 of 175 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_banking_pain_base
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_banking_pain_base/es/

alexis-via committed:
Large refactoring/improvement/cleanup of OCA/bank-payment

NEW FEATURES
- activate 'in_payment' payment_state on invoices... at last !
- payment order numbering: add ability to have a sequence specific to a particular payment mode (optional)
- add show account number policy "first_last" to show N first and N last characters of bank account number

SMALL IMPROVEMENTS
- account.payment.method: rename field "payment_order_only" to "payment_order_ok" for naming consistency with the field of account.payment.mode. Auto-set "payment_order_ok" of payment mode from the equivalent field of payment method. Migration scripts provided for account.payment.method and the data of SEPA credit transfer and SEPA direct debit.
- payment order attachment: add M2O payment_file_id that points to the payment file. No more need for a simplified attachment form view.
- keep the spaces in the scrambled bank account number, so that the user easily understands that it is a "scrambled" bank account number
- add field acc_number_scrambled on res.partner.bank for easy and direct use of scrambled account number

SIMPLIFICATION / CODE CLEANUP
- remove field partner_bank_id on account.move.line
- move code for scrambled account number from qweb report to res.partner.bank class
- remove boolean field show_bank_account_from_journal from account.payment.mode: we now consider that this option is always true i.e. when account_payment_partner is installed, Odoo takes the bank accounts to display on the customer invoice from the payment mode (if the payment mode is empty, it displays all the bank accounts of the company). As a consequence, hide partner_bank_id on customer invoice form view.
- remove fields inbound_payment_order_only and outbound_payment_order_only from account.journal : these 2 fields were not used!
- add check_company=True where it was missing and remove code of constraint that checks company consistency
- re-organise form view of payment order to equilibrate between left column and right column

TECHNICAL:
- replace @api.onchange by computed readonly=False store=True fields
- remove (or reduce) the use of demo data in tests
- speed-up tests by using tracking_disable=True
- Replace (6, 0, []) and (0, 0, {}) by Command.set() and Command.create()
- add sql unicity constraint on payment order number per company

alexis-via committed:
Add support for regulatory reporting in PAIN files

Harmonize values for reference_type on account.move and communication_type on account.payment.line:
2 possible values : free and structured (migration script provided)
Simplify code for reference.

alexis-via committed:
account_banking_pain_base: use _rec_names_search for account.pain.regulatory.reporting

alexis-via committed:
Big cleanup of PAIN XML code generation

Use lxml.objectify instead of lxml.etree for XML generation
Add support for currencies with decimal places != 2
Stop using safe_eval() in _prepare_field()
Add arg 'raise_if_oversized' in _prepare_field() to avoid cutting fields
that should never be cut
More robust implementation of allowed characters in _prepare_field():
filter on the allower characters instead of filtering in unallowed
characters
Don't use _prepare_field() where it is not really needed (on dates,
country code, currency code)
Remove support for pain.001.001.02: nobody uses it (pain.001.001.03 must
be used according to EPC) and it adds maintenance burden
Switch methods from public to private where relevant
Remove noqa: C901 in account_banking_sepa_credit_transfer
Add tests in account_banking_pain_base

alexis-via committed:
account_banking_pain_base: structured address in XML

EPC says that the use of structured addresses will be mandatory starting
November 2025.

alexis-via committed:
[IMP] add support for PAIN 001.001.09 + 008.001.08 (adopted by EPC)

Remove support for:
- SCT : pain.001.001.04, pain.001.001.05
- SDD : ain.008.001.03, ain.008.001.04
because these PAIN version have never been adopted by EPC.
Update and improve README

alexis-via committed:
account_banking_pain_base: hide pain_version in payment method views for non sepa payment method

alexis-via committed:
account_banking_pain_base: change group for full-access ACL for account.pain.regulatory.reporting

alexis-via committed:
[FIX] Layout of SEPA section of accounting config page

alexis-via committed:
[IMP] add related field payment_method_code on account.move and account.payment.order

This new field payment_method_code on account.move and
account.payment.order is available as invisible in tree+form view: it
allows to easily show/hide fields specific to a payment method in format-specific modules.
Remove field sepa_payment_method (replaced by generic field payment_method_code)
Update field definition of payment_mode_id on account.move: precompute=True and move domain from view to field definition

alexis-via committed:
[IMP] account_banking_pain_base: pre-commit auto fixes

alexis-via committed:
[MIG] account_banking_pain_base to v18

alexis-via committed:
Rename account_banking_pain_base to account_payment_sepa_base

Update manifest and readme

alexis-via committed:
[FIX] account_payment_sepa_base: fix website in manifest

alexis-via commented:
It seems that the test "Detect unreleased dependencies" is unable to properly read test-requirements.txt, whereas other tests read it fine.
@sbidoul do you have a clue about this ?

sbidoul commented:
@alexis-via all is working as expected here. This test is failing as expected to prevent merging this PR until its dependencies are merged.

alexis-via committed:
[IMP] account_payment_sepa_base: add support for account.payment.lot

FIX XMLID for group in res.config.settings page

Add hook for sepa computation on payment lines (used in account_payment_sepa_direct_debit module)

Remove field convert_to_ascii on account.payment.method, because it
must never be disabled for SEPA payments and is very rarely disabled for
non-sepa payments. Replaced by a hook for those who really need it.

Add prepare method dedicated to filename

Move method _generate_start_payment_info_block to account.payment.lot-
with several improvements/cleanup.

Rename file security/security.xml to security/res_groups.xml

alexis-via committed:
[IMP] account_payment_sepa_base: pre-commit update

alexis-via commented:
ouf, green at last !

alexis-via committed:
[UDP] account_payment_sepa_base: remove prepare method for filename which is now part of account_payment_batch_oca

alexis-via committed:
[IMP] account_payment_sepa_base: improve strings

Fixes #18

sbidoul commented:
/ocabot merge nobump
